src/routing_core.rs

```diff
@@ -268,8 +268,12 @@ impl RoutingCore {
                                                               vec![endpoint.clone()],
                                                               Some(connection));
                                 let routing_table_count_prior = routing_table.size();
-                                // TODO (ben 10/08/2015) drop connection of dropped node
-                                let (added, _) = routing_table.add_node(node_info);
+                                let (added, previous) = routing_table.add_node(node_info);
```

Comment:
the `previous` node needs to be dropped in the crust::connection_manager; it will already be removed from the routing table.  This needs to be passed back over the Action channel to routing_node to drop this connection